src/dr_llm/llm/providers/anthropic/reasoning.py

```
@@ -5,15 +5,32 @@
     ThinkingLevel,
 )
 
 
+def _require_anthropic_api_budget_tokens(
+    budget_tokens: int | None, *, label: str
+) -> int:
+    """Validate tokens for Anthropic-style thinking payloads (strictly positive int)."""
+    if budget_tokens is None:
+        raise ValueError(
+            f"{label} budget thinking requires budget_tokens when thinking_level is 'budget'"
+        )
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)